operator/api/v1alpha2/seldondeployment_types.go

```diff
@@ -0,0 +1,372 @@
+	OauthSecret string            `json:"oauth_secret,omitempty" protobuf:"string,4,opt,name=oauth_secret"`
+	Annotations map[string]string `json:"annotations,omitempty" protobuf:"bytes,5,opt,name=annotations"`
+}
+
+type PredictorSpec struct {
+	Name            string                  `json:"name,omitempty" protobuf:"string,1,opt,name=name"`
+	Graph           *PredictiveUnit         `json:"graph,omitempty" protobuf:"bytes,2,opt,name=predictiveUnit"`
+	ComponentSpecs  []*SeldonPodSpec        `json:"componentSpecs,omitempty" protobuf:"bytes,3,opt,name=componentSpecs"`
+	Replicas        int32                   `json:"replicas,omitempty" protobuf:"string,4,opt,name=replicas"`
+	Annotations     map[string]string       `json:"annotations,omitempty" protobuf:"bytes,5,opt,name=annotations"`
+	EngineResources v1.ResourceRequirements `json:"engineResources,omitempty" protobuf:"bytes,6,opt,name=engineResources"`
+	Labels          map[string]string       `json:"labels,omitempty" protobuf:"bytes,7,opt,name=labels"`
+	SvcOrchSpec     SvcOrchSpec             `json:"svcOrchSpec,omitempty" protobuf:"bytes,8,opt,name=svcOrchSpec"`
+	Traffic         int32                   `json:"traffic,omitempty" protobuf:"bytes,9,opt,name=traffic"`
+	Explainer       Explainer               `json:"explainer,omitempty" protobuf:"bytes,10,opt,name=explainer"`
+}
+
+type SvcOrchSpec struct {
+	Resources *v1.ResourceRequirements `json:"resources,omitempty" protobuf:"bytes,1,opt,name=resources"`
+	Env       []*v1.EnvVar             `json:"env,omitempty" protobuf:"bytes,2,opt,name=env"`
+}
+
+type Explainer struct {
+	Type               string            `json:"type,omitempty" protobuf:"string,1,opt,name=type"`
+	ModelUri           string            `json:"modelUri,omitempty" protobuf:"string,2,opt,name=modelUri"`
+	ServiceAccountName string            `json:"serviceAccountName,omitempty" protobuf:"string,3,opt,name=serviceAccountName"`
+	ContainerSpec      v1.Container      `json:"containerSpec,omitempty" protobuf:"bytes,4,opt,name=containerSpec"`
+	Config             map[string]string `json:"config,omitempty" protobuf:"bytes,5,opt,name=config"`
+	Endpoint           *Endpoint         `json:"endpoint,omitempty" protobuf:"bytes,6,opt,name=endpoint"`
+	EnvSecretRefName   string            `json:"envSecretRefName,omitempty" protobuf:"bytes,7,opt,name=envSecretRefName"`
+}
+
+type SeldonPodSpec struct {
+	Metadata metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	Spec     v1.PodSpec        `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+	HpaSpec  *SeldonHpaSpec    `json:"hpaSpec,omitempty" protobuf:"bytes,3,opt,name=hpaSpec"`
+}
+
+type SeldonHpaSpec struct {
+	MinReplicas *int32                          `json:"minReplicas,omitempty" protobuf:"int,1,opt,name=minReplicas"`
+	MaxReplicas int32                           `json:"maxReplicas,omitempty" protobuf:"int,2,opt,name=maxReplicas"`
```

Comment:
MinReplicas and Maxreplicas also use different value/reference.

Reply:
also, for the value omitempty can cause other issues.  Is there a reason why almost every field is omitempty?  That makes validation solely the responsibility of the validating webhookl.

Reply:
yes - remove omitempy